diff --git browser/base/content/baseMenuOverlay.xul browser/base/content/baseMenuOverlay.xul
index 804bbd..320693 100644
--- a/browser/base/content/baseMenuOverlay.xul
+++ b/browser/base/content/baseMenuOverlay.xul
@@ -38,8 +38,7 @@
           accesskey="&helpMenuWin.accesskey;">
 #else
     <menu id="helpMenu"
-          label="&helpMenu.label;"
-          accesskey="&helpMenu.accesskey;">
+          label="&helpMenu.label;">
 #endif
       <menupopup id="menu_HelpPopup" onpopupshowing="buildHelpMenu();">
         <menuitem id="menu_openHelp"
diff --git browser/base/content/browser-menubar.inc browser/base/content/browser-menubar.inc
index 6dc75f..fc229e 100644
--- a/browser/base/content/browser-menubar.inc
+++ b/browser/base/content/browser-menubar.inc
@@ -9,8 +9,7 @@
                                   this.setAttribute('openedwithkey',
                                                     event.target.parentNode.openedWithKey);"
                 style="border:0px;padding:0px;margin:0px;-moz-appearance:none">
-            <menu id="file-menu" label="&fileMenu.label;"
-                  accesskey="&fileMenu.accesskey;">
+            <menu id="file-menu" label="&fileMenu.label;">
               <menupopup id="menu_FilePopup"
                          onpopupshowing="updateUserContextUIVisibility();">
                 <menuitem id="menu_newNavigatorTab"
@@ -118,8 +117,7 @@
               </menupopup>
             </menu>
 
-            <menu id="edit-menu" label="&editMenu.label;"
-                  accesskey="&editMenu.accesskey;">
+            <menu id="edit-menu" label="&editMenu.label;">
               <menupopup id="menu_EditPopup"
                          onpopupshowing="updateEditUIVisibility()"
                          onpopuphidden="updateEditUIVisibility()">
@@ -191,8 +189,7 @@
               </menupopup>
             </menu>
 
-            <menu id="view-menu" label="&viewMenu.label;"
-                  accesskey="&viewMenu.accesskey;">
+            <menu id="view-menu" label="&viewMenu.label;">
               <menupopup id="menu_viewPopup"
                          onpopupshowing="updateCharacterEncodingMenuState();">
                 <menu id="viewToolbarsMenu"
@@ -317,8 +314,7 @@
             </menu>
 
             <menu id="history-menu"
-                  label="&historyMenu.label;"
-                  accesskey="&historyMenu.accesskey;">
+                  label="&historyMenu.label;">
               <menupopup id="goPopup"
 #ifndef XP_MACOSX
                          placespopup="true"
@@ -375,7 +371,6 @@
 
   <menu id="bookmarksMenu"
         label="&bookmarksMenu.label;"
-        accesskey="&bookmarksMenu.accesskey;"
         ondragenter="PlacesMenuDNDHandler.onDragEnter(event);"
         ondragover="PlacesMenuDNDHandler.onDragOver(event);"
         ondrop="PlacesMenuDNDHandler.onDrop(event);">
@@ -469,8 +464,7 @@
   </menu>
 
             <menu id="tools-menu"
-                  label="&toolsMenu.label;"
-                  accesskey="&toolsMenu.accesskey;">
+                  label="&toolsMenu.label;">
               <menupopup id="menu_ToolsPopup">
               <menuitem id="menu_openDownloads"
                         label="&downloads.label;"
diff --git browser/base/content/tabbrowser.xml browser/base/content/tabbrowser.xml
index 4f40e4..88fd68 100644
--- a/browser/base/content/tabbrowser.xml
+++ b/browser/base/content/tabbrowser.xml
@@ -5438,25 +5438,25 @@
             return;
           }
 
-          if (aEvent.altKey)
-            return;
-
           // Don't check if the event was already consumed because tab
           // navigation should always work for better user experience.
 
-          if (aEvent.ctrlKey && aEvent.shiftKey && !aEvent.metaKey) {
+          if (!aEvent.ctrlKey && aEvent.shiftKey && aEvent.altKey && !aEvent.metaKey) {
             switch (aEvent.keyCode) {
-              case aEvent.DOM_VK_PAGE_UP:
+              case aEvent.DOM_VK_H:
                 this.moveTabBackward();
                 aEvent.preventDefault();
                 return;
-              case aEvent.DOM_VK_PAGE_DOWN:
+              case aEvent.DOM_VK_L:
                 this.moveTabForward();
                 aEvent.preventDefault();
                 return;
             }
           }
 
+          if (aEvent.altKey)
+            return;
+
           if (AppConstants.platform != "macosx") {
             if (aEvent.ctrlKey && !aEvent.shiftKey && !aEvent.metaKey &&
                 aEvent.keyCode == KeyEvent.DOM_VK_F4 &&
diff --git toolkit/content/widgets/tabbox.xml toolkit/content/widgets/tabbox.xml
index debaf3..035a86 100644
--- a/toolkit/content/widgets/tabbox.xml
+++ b/toolkit/content/widgets/tabbox.xml
@@ -150,17 +150,17 @@
           // navigation should always work for better user experience.
 
           switch (event.keyCode) {
-            case event.DOM_VK_TAB:
-              if (event.ctrlKey && !event.altKey && !event.metaKey)
+            case event.DOM_VK_H:
+              if (!event.ctrlKey && !event.shiftKey && event.altKey && !event.metaKey)
                 if (this.tabs && this.handleCtrlTab) {
-                  this.tabs.advanceSelectedTab(event.shiftKey ? -1 : 1, true);
+                  this.tabs.advanceSelectedTab(-1, true);
                   event.preventDefault();
                 }
               break;
-            case event.DOM_VK_PAGE_UP:
-              if (event.ctrlKey && !event.shiftKey && !event.altKey && !event.metaKey)
+            case event.DOM_VK_L:
+              if (!event.ctrlKey && !event.shiftKey && event.altKey && !event.metaKey)
                 if (this.tabs && this.handleCtrlPageUpDown) {
-                  this.tabs.advanceSelectedTab(-1, true);
+                  this.tabs.advanceSelectedTab(1, true);
                   event.preventDefault();
                 }
               break;
