From e8664c6c0f2d983263c610cd968a2232e4037396 Mon Sep 17 00:00:00 2001
From: deso <deso@devnull.com>
Date: Sat, 23 Feb 2013 22:06:48 -0800
Subject: [PATCH 6/6] added shortcut for destroying a tab

---
 src/perl/tabbed | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/perl/tabbed b/src/perl/tabbed
index 43ca1f4..a593f14 100644
--- a/src/perl/tabbed
+++ b/src/perl/tabbed
@@ -381,6 +381,7 @@ sub tab_key_press {
    my $key_h = 0x0068; # small letter h
    my $key_l = 0x006c; # small letter l
    my $key_n = 0x006e; # small letter n
+   my $key_w = 0x0077; # small letter w
    my $key_H = 0x0048; # capital letter H
    my $key_L = 0x004c; # capital letter L
 
@@ -445,6 +446,13 @@ sub tab_key_press {
          }
          return 1;
       }
+      elsif ($keysym == $key_w)
+      {
+         # we want to actually stop the process, not just destroy the tab
+         # for whatever reason sending SIGTERM (15) does not work (to shutdown gracefully), so we
+         # have to use SIGKILL (9) to do it the hard way
+         kill 9, $self->{cur}->get_pid();
+      }
       elsif (exists $tab_index{$keysym})
       {
          my $idx = $tab_index{$keysym};
-- 
2.0.4

