From c902df845a70140cc42fe80b7c92813a9f03cc12 Mon Sep 17 00:00:00 2001
From: deso <deso@devnull.com>
Date: Sat, 23 Feb 2013 08:35:29 -0800
Subject: [PATCH 5/6] fixed tab focusing problem

---
 src/perl/tabbed | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/perl/tabbed b/src/perl/tabbed
index 6511474..43ca1f4 100644
--- a/src/perl/tabbed
+++ b/src/perl/tabbed
@@ -338,12 +338,13 @@ sub tab_start {
 sub tab_destroy {
    my ($self, $tab) = @_;
 
+   my ($idx) = grep $self->{tabs}[$_] == $tab, 0 .. $#{ $self->{tabs} };
    $self->{tabs} = [ grep $_ != $tab, @{ $self->{tabs} } ];
 
    if (@{ $self->{tabs} }) {
       if ($self->{cur} == $tab) {
          delete $self->{cur};
-         $self->make_current ($self->{tabs}[-1]);
+         $self->make_current ($self->{tabs}[$idx < $#{ $self->{tabs} } ? $idx : -1]);
       } else {
          $self->refresh;
       }
-- 
2.0.4

