diff --git src/command.c src/command.c
index 53af95..9ddc8c 100644
--- a/src/command.c
+++ b/src/command.c
@@ -342,7 +342,7 @@ find_songname_id(struct mpd_connection *conn, const char *s)
 	mpd_search_queue_songs(conn, false);
 
 	const char *pattern = charset_to_utf8(s);
-	mpd_search_add_any_tag_constraint(conn, MPD_OPERATOR_DEFAULT,
+	mpd_search_add_uri_constraint(conn, MPD_OPERATOR_DEFAULT,
 					       pattern);
 	mpd_search_commit(conn);
 
