diff --git src/command.c src/command.c
index 31780d..fed0dd 100644
--- a/src/command.c
+++ b/src/command.c
@@ -336,7 +336,7 @@ find_songname_id(struct mpd_connection *conn, int argc, char **argv)
 
 	if (argc == 1) {
 		const char *pattern = charset_to_utf8(argv[0]);
-		mpd_search_add_any_tag_constraint(conn, MPD_OPERATOR_DEFAULT,
+	        mpd_search_add_uri_constraint(conn, MPD_OPERATOR_DEFAULT,
 						  pattern);
 	} else {
 		int n = add_constraints(argc, argv, conn);
