diff --git src/imports.rs src/imports.rs
index 3b7c47..8e4cbb 100644
--- a/src/imports.rs
+++ b/src/imports.rs
@@ -29,7 +29,7 @@ fn path_of(a: &ast::ViewPath_) -> &ast::Path {
 }
 
 fn compare_path_segments(a: &ast::PathSegment, b: &ast::PathSegment) -> Ordering {
-    a.identifier.name.as_str().cmp(&b.identifier.name.as_str())
+    a.identifier.name.as_str().to_lowercase().cmp(&b.identifier.name.as_str().to_lowercase())
 }
 
 fn compare_paths(a: &ast::Path, b: &ast::Path) -> Ordering {
