diff --git a/lib/portage/util/env_update.py b/lib/portage/util/env_update.py
index d76042..d32474 100644
--- a/lib/portage/util/env_update.py
+++ b/lib/portage/util/env_update.py
@@ -395,9 +395,9 @@ def _env_update(makelinks, target_root, prev_mtimes, contents, env, writemsg_lev
         for k in env_keys:
             v = env[k]
             if v.startswith("$") and not v.startswith("${"):
-                outfile.write(f"export {k}=$'{v[1:]}'\n")
+                outfile.write(f"export {k}=$\"{v[1:]}\"\n")
             else:
-                outfile.write(f"export {k}='{v}'\n")
+                outfile.write(f"export {k}=\"{v}\"\n")
 
     # Create the systemd user environment configuration file
     # /etc/environment.d/10-gentoo-env.conf with the
