--- a/lib/portage/util/env_update.py	2020-04-11 17:57:47.000000000 -0700
+++ b/lib/portage/util/env_update.py	2020-05-10 16:31:55.600007473 -0700
@@ -352,9 +352,9 @@
 	for k in env_keys:
 		v = env[k]
 		if v.startswith('$') and not v.startswith('${'):
-			outfile.write("export %s=$'%s'\n" % (k, v[1:]))
+			outfile.write("export %s=$\"%s\"\n" % (k, v[1:]))
 		else:
-			outfile.write("export %s='%s'\n" % (k, v))
+			outfile.write("export %s=\"%s\"\n" % (k, v))
 	outfile.close()
 
 	#create /etc/csh.env for (t)csh support
