diff --git src/help.txt src/help.txt
index 588755..bd2fd2 100644
--- src/help.txt
+++ src/help.txt
@@ -145,10 +145,10 @@ through the file.
   g/home            Move to the top of the file.
   G/end             Move to the end of the file.  If the view is already
                     at the end, it will move to the last line.
-  space/pgdn        Move down a page.
-  b/bs/pgup         Move up a page.
-  j/cr/down-arrow   Move down a line.
-  k/up-arrow        Move up a line.
+  space/pgdn/CTRL+d Move down a page.
+  b/bs/pgup/CTRL+u  Move up a page.
+  j/cr/down-arrow/CTRL+e Move down a line.
+  k/up-arrow/CTRL+y Move up a line.
   h/left-arrow      Move to the left.
   l/right-arrow     Move to the right.
 
diff --git src/listview_curses.cc src/listview_curses.cc
index 33ebca..43fdb6 100644
--- src/listview_curses.cc
+++ src/listview_curses.cc
@@ -99,22 +99,31 @@ bool listview_curses::handle_key(int ch)
     case '\r':
     case 'j':
     case KEY_DOWN:
+    /*
+     * VIM's semantic of Ctrl+e is slightly different from j or key down
+     * in that it only moves the view, not the cursor, but since we do
+     * not have a cursor here the binding accomplishes the same task.
+     */
+    case KEY_CTRL_E:
         this->shift_top(vis_line_t(1));
         break;
 
     case 'k':
     case KEY_UP:
+    case KEY_CTRL_Y:
         this->shift_top(vis_line_t(-1));
         break;
 
     case 'b':
     case KEY_BACKSPACE:
     case KEY_PPAGE:
+    case KEY_CTRL_U:
         this->shift_top(-(this->rows_available(this->lv_top, RD_UP) - vis_line_t(1)));
         break;
 
     case ' ':
     case KEY_NPAGE:
+    case KEY_CTRL_D:
         this->shift_top(this->rows_available(this->lv_top, RD_DOWN) - vis_line_t(1));
         break;
 
diff --git src/lnav.cc src/lnav.cc
index c14546..5d9043 100644
--- src/lnav.cc
+++ src/lnav.cc
@@ -3378,7 +3378,6 @@ private:
 static void handle_key(int ch)
 {
     switch (ch) {
-    case CEOF:
     case KEY_RESIZE:
         break;
     default:
@@ -3668,7 +3667,6 @@ static void looper(void)
                             continue;
                         }
                         switch (ch) {
-                        case CEOF:
                         case KEY_RESIZE:
                             break;
 
diff --git src/view_curses.hh src/view_curses.hh
index 6ffbed..868fd2 100644
--- src/view_curses.hh
+++ src/view_curses.hh
@@ -62,9 +62,13 @@
 
 #include "lnav_log.hh"
 
+#define KEY_CTRL_D    4
+#define KEY_CTRL_E    5
 #define KEY_CTRL_G    7
 #define KEY_CTRL_R    18
+#define KEY_CTRL_U    21
 #define KEY_CTRL_W    23
+#define KEY_CTRL_Y    25
 #define KEY_CTRL_RBRACKET 0x1d
 
 class view_curses;
