diff --git a/src/tools/rls/rls/src/build/cargo.rs b/src/tools/rls/rls/src/build/cargo.rs
--- a/src/tools/rls/rls/src/build/cargo.rs
+++ b/src/tools/rls/rls/src/build/cargo.rs
@@ -269,7 +269,7 @@ fn run_cargo_ws(
     // target/ existing already and won't exclude it from backups. We can work around that by
     // attempting to create a backup-excluded target/ ourelves using cargo paths:: machinery.
     cargo::util::paths::create_dir_all_excluded_from_backups_atomic(
-        config.target_dir().unwrap().unwrap().as_path_unlocked().parent().unwrap(),
+        config.target_dir(manifest_path).unwrap().unwrap().as_path_unlocked().parent().unwrap(),
     )?;
     let exec = Arc::new(exec) as Arc<dyn Executor>;
     match compile_with_exec(&ws, &compile_opts, &exec) {
