--- a/runtime/lua/vim/lsp/inlay_hint.lua
+++ b/runtime/lua/vim/lsp/inlay_hint.lua
@@ -346,7 +346,9 @@
         end
 
         for pos, vt in pairs(hint_virtual_texts) do
-          api.nvim_buf_set_extmark(bufnr, namespace, lnum, pos, {
+          local line = api.nvim_buf_get_lines(bufnr, lnum, lnum + 1, false)[1]
+          local col = math.min(pos, line and #line or 0)
+          api.nvim_buf_set_extmark(bufnr, namespace, lnum, col, {
             virt_text_pos = 'inline',
             ephemeral = false,
             virt_text = vt,
