diff --git vim74/runtime/indent/python.vim vim74/runtime/indent/python.vim
index 4ce3c23..af91b66 100644
--- vim74/runtime/indent/python.vim
+++ vim74/runtime/indent/python.vim
@@ -91,7 +91,7 @@ function GetPythonIndent(lnum)
       if pp > 0
 	return indent(plnum) + (exists("g:pyindent_nested_paren") ? eval(g:pyindent_nested_paren) : shiftwidth())
       endif
-      return indent(plnum) + (exists("g:pyindent_open_paren") ? eval(g:pyindent_open_paren) : (shiftwidth() * 2))
+      return indent(plnum) + (exists("g:pyindent_open_paren") ? eval(g:pyindent_open_paren) : shiftwidth())
     endif
     if plnumstart == p
       return indent(plnum)
