From 7a31e559c6e08e5b59670320359b44f86b798a55 Mon Sep 17 00:00:00 2001
From: deso <deso@devnull.com>
Date: Sat, 23 Feb 2013 08:32:03 -0800
Subject: [PATCH 2/6] added custom paste shortcuts

---
 src/command.C | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/src/command.C b/src/command.C
index 153f0e1..7f659c5 100644
--- a/src/command.C
+++ b/src/command.C
@@ -604,6 +604,28 @@ rxvt_term::key_press (XKeyEvent &ev)
           return;
         }
 
+      if (ctrl && keysym == XK_v)
+      {
+        // Ctrl+V = paste mouse selection
+        selection_request (ev.time, Sel_Primary);
+        return;
+      }
+      else if (ctrl && shft && keysym == XK_V)
+      {
+        // Ctrl+Shift+V = paste clipboard
+        selection_request (ev.time, Sel_Clipboard);
+        return;
+      }
+      /**
+       * @todo this does not work! we need a way to copy something into the "real" clipboard
+       */
+      //else if (ctrl && shft && keysym == XK_C)
+      //{
+      //  // Ctrl+Shift+C = copy selection to clipboard
+      //  selection_make (ev.time);
+      //  return;
+      //}
+
 #if ENABLE_FRILLS || ISO_14755
       // ISO 14755 support
       if (iso14755buf & (ISO_14755_STARTED | ISO_14755_51))
-- 
2.0.4

