From 1e439139aaf1b1faae795258896802096fc7c894 Mon Sep 17 00:00:00 2001
From: deso <deso@devnull.com>
Date: Sat, 23 Feb 2013 08:32:46 -0800
Subject: [PATCH 3/6] enabled "scoll on up/down keys" feature

---
 src/feature.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/feature.h b/src/feature.h
index 151d29e..682d63b 100644
--- a/src/feature.h
+++ b/src/feature.h
@@ -117,7 +117,7 @@
  * SCROLL_ON_* defines above.).
  * Also for modifier+Home/End keys to move to top/bottom
  */
-/* #define SCROLL_ON_UPDOWN_KEYS */
+#define SCROLL_ON_UPDOWN_KEYS
 /* #define SCROLL_ON_HOMEEND_KEYS */
 
 /*
-- 
2.0.4

